crates/sui-graphql-rpc/src/types/epoch.rs

```diff
@@ -185,8 +260,12 @@ impl Epoch {
     }
 }
 
-impl From<StoredEpochInfo> for Epoch {
-    fn from(e: StoredEpochInfo) -> Self {
-        Epoch { stored: e }
+pub fn from_epoch_and_system_state(
+    stored: StoredEpochInfo,
+    system_state: NativeSuiSystemStateSummary,
+) -> Epoch {
+    Epoch {
+        stored,
+        system_state,
     }
 }
```

Comment:
I know I mentioned generally preferring free functions if you don't access `&self` but constructors are an exception to that rule -- I think it's fair game to make this an `impl Epoch { fn from(stored: StoredEpochInfo, system_state: NativeSuiSystemStateSummary) { /* ... */ } }`

But based on my earlier comment about making the system state summary computed, you may not need to make this change.